ui/src/shared/components/DatabaseDropdown.js

```diff
@@ -43,12 +43,18 @@ class DatabaseDropdown extends Component {
     const proxy = source.links.proxy
     try {
       const {data} = await showDatabases(proxy)
-      const {databases} = showDatabasesParser(data)
+      const {databases, errors} = showDatabasesParser(data)
+      if (errors.length > 0) {
+        throw errors[0] // only one error can come back from this, but it's returned as an array
+      }
 
-      this.setState({databases})
-      const selectedDatabaseText = databases.includes(database)
+      // system databases are those preceded with an '_'
+      const nonSystemDatabases = databases.filter((name) => name[0] !== '_')
```

Comment:
You can use the ES6 `.startsWith` here instead of `name[0]`. It'll read more semantically.